controller_manager/include/controller_manager/controller_manager.hpp

```diff
@@ -69,44 +84,211 @@ class ControllerManager : public rclcpp::Node
     typename T,
     typename std::enable_if<std::is_convertible<
       T *, controller_interface::ControllerInterface *>::value, T>::type * = nullptr>
-  std::shared_ptr<controller_interface::ControllerInterface>
-  add_controller(std::shared_ptr<T> controller, std::string controller_name)
+  controller_interface::ControllerInterfaceSharedPtr
+  add_controller(
+    std::shared_ptr<T> controller, const std::string & controller_name,
+    const std::string & controller_type)
   {
-    return add_controller_impl(controller, controller_name);
+    ControllerSpec controller_spec;
```

Comment:
Moved to a new issue since we'll need to modify the existing controllers as well in the other repo and don't want to drag this PR longer than necessary. https://github.com/ros-controls/ros2_control/issues/162